android/src/main/java/com/polidea/flutter_ble_lib/delegate/DeviceConnectionDelegate.java

```diff
@@ -97,13 +101,25 @@ public void onError(BleError error) {
                     public void onSuccess(Device data) {
                         safeMainThreadResolver.onSuccess(null);
                     }
-                }, new OnEventCallback<ConnectionState>() {
+                },
+                new OnEventCallback<ConnectionState>() {
                     @Override
                     public void onEvent(final ConnectionState data) {
+                        String temp;
+                        try {
+                            temp = new ConnectionStateChangeJsonConverter().toJson(new ConnectionStateChange(deviceId, data));
```

Comment:
There is no need to create json converter instances on every event. We can declare this converter as local for this method.